Makefile

```diff
@@ -20,7 +20,7 @@ include $(ROOT_DIR_RELATIVE)/common.mk
 # https://suva.sh/posts/well-documented-makefiles
 
 # Go
-GO_VERSION ?=1.23.9
+GO_VERSION ?=1.24.0
```

Comment:
Let's use the latest go 1.24 in this line

Reply:
I think that's because they use it for verification reasons (see [here](https://github.com/kubernetes-sigs/cluster-api/blob/1d9f42a989db9c9d5da30633b68d520450bfa94e/Makefile#L698-L699))

They are still setting it manually in the go.mod, like we do https://github.com/kubernetes-sigs/cluster-api/blob/1d9f42a989db9c9d5da30633b68d520450bfa94e/go.mod#L3